### ISSUE TITLE
[🐛 Bug]: Custom strategy arguments not provided when refetching a stale element

### ISSUE THREAD

**Issue by chrisbottin** [webdriverio/webdriverio#8176]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

7.19.1

### Node.js Version

v16.14.0

### Mode

WDIO Testrunner

### What happened?

When using a custom strategy to find an element and the element becomes stale before interacting with it again, the original custom strategy arguments are not provided to the custom strategy function which causes a bug. 

### What is your expected behavior?

When using a custom strategy and the element becomes stale, the refetch functionality should correctly find and return the element again.

### How to reproduce the bug.

Here is a simple test to reproduce the bug ...
```js
it('custom strategy bug on refetch', async function() {
    await browser.addLocatorStrategy('getLinkByHref', (href) => {
        if (!href) {
            throw new Error('"href" parameter is required');
        }
        return document.querySelector(`a[href='${href}']`) as any;
    });

    await browser.url('https://webdriver.io');
    const link = await browser.custom$('getLinkByHref', '/docs/api');

    // Refresh the page to make the link element stale
    await browser.refresh();

    // This method fails during the refetch of the element, the error "'href' parameter is required" is thrown.
    await link.click();

    await expect(browser).toHaveTitle('Introduction | WebdriverIO');
});
```

### Relevant log output

```typescript
[1648808029.919][INFO]: Starting ChromeDriver 100.0.4896.60 (6a5d10861ce8de5fce22564658033b43cb7de047-refs/branch-heads/4896@{#875}) on port 65077
[1648808029.919][INFO]: Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
[1648808030.496][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND InitSession {
   "capabilities": {
      "alwaysMatch": {
         "acceptInsecureCerts": true,
         "browserName": "chrome"
      },
      "firstMatch": [ {

      } ]
   },
   "desiredCapabilities": {
      "acceptInsecureCerts": true,
      "browserName": "chrome"
   }
}
[1648808030.500][INFO]: Populating Preferences file: {
   "alternate_error_pages": {
      "enabled": false
   },
   "autofill": {
      "enabled": false
   },
   "browser": {
      "check_default_browser": false
   },
   "distribution": {
      "import_bookmarks": false,
      "import_history": false,
      "import_search_engine": false,
      "make_chrome_default_for_user": false,
      "skip_first_run_ui": true
   },
   "dns_prefetching": {
      "enabled": false
   },
   "profile": {
      "content_settings": {
         "pattern_pairs": {
            "https://*,*": {
               "media-stream": {
                  "audio": "Default",
                  "video": "Default"
               }
            }
         }
      },
      "default_content_setting_values": {
         "geolocation": 1
      },
      "default_content_settings": {
         "geolocation": 1,
         "mouselock": 1,
         "notifications": 1,
         "popups": 1,
         "ppapi-broker": 1
      },
      "password_manager_enabled": false
   },
   "safebrowsing": {
      "enabled": false
   },
   "search": {
      "suggest_enabled": false
   },
   "translate": {
      "enabled": false
   }
}
[1648808030.500][INFO]: Populating Local State file: {
   "background_mode": {
      "enabled": false
   },
   "ssl": {
      "rev_checking": {
         "enabled": false
      }
   }
}
[1648808030.501][INFO]: Launching chrome: /Applications/Google Chrome.app/Contents/MacOS/Google Chrome --allow-pre-commit-input --disable-background-networking --disable-client-side-phishing-detection --disable-default-apps --disable-hang-monitor --disable-popup-blocking --disable-prompt-on-repost --disable-sync --enable-automation --enable-blink-features=ShadowDOMV0 --enable-logging --ignore-certificate-errors --log-level=0 --no-first-run --no-service-autorun --password-store=basic --remote-debugging-port=0 --test-type=webdriver --use-mock-keychain --user-data-dir=/var/folders/w1/k5z8hmrj4wq8v29_n5_hxl2xh68qx2/T/.com.google.Chrome.fzCkw6 data:,
[1648808032.277][INFO]: resolved localhost to ["::1","127.0.0.1"]
[1648808032.298][INFO]: resolved localhost to ["::1","127.0.0.1"]
[1648808032.319][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE InitSession {
   "capabilities": {
      "acceptInsecureCerts": true,
      "browserName": "chrome",
      "browserVersion": "100.0.4896.60",
      "chrome": {
         "chromedriverVersion": "100.0.4896.60 (6a5d10861ce8de5fce22564658033b43cb7de047-refs/branch-heads/4896@{#875})",
         "userDataDir": "/var/folders/w1/k5z8hmrj4wq8v29_n5_hxl2xh68qx2/T/.com.google.Chrome.fzCkw6"
      },
      "goog:chromeOptions": {
         "debuggerAddress": "localhost:65083"
      },
      "networkConnectionEnabled": false,
      "pageLoadStrategy": "normal",
      "platformName": "mac os x",
      "proxy": {

      },
      "setWindowRect": true,
      "strictFileInteractability": false,
      "timeouts": {
         "implicit": 0,
         "pageLoad": 300000,
         "script": 30000
      },
      "unhandledPromptBehavior": "dismiss and notify",
      "webauthn:extension:credBlob": true,
      "webauthn:extension:largeBlob": true,
      "webauthn:virtualAuthenticators": true
   },
   "sessionId": "74e99ee11edc2373110d8d6076d1bce3"
}
[1648808032.356][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND MaximizeWindow {

}
[1648808032.356][INFO]: Waiting for pending navigations...
[1648808032.358][INFO]: Done waiting for pending navigations. Status: ok
[1648808032.601][INFO]: Waiting for pending navigations...
[1648808032.603][INFO]: Done waiting for pending navigations. Status: ok
[1648808032.603][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE MaximizeWindow {
   "height": 930,
   "width": 1680,
   "x": 0,
   "y": 25
}
[1648808032.677][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND GetWindow {

}
[1648808032.677][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE GetWindow "CDwindow-B7EC9AE5DF32A499CA76B15506C5E17A"
[1648808032.696][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND Navigate {
   "url": "https://webdriver.io/"
}
[1648808032.696][INFO]: Waiting for pending navigations...
[1648808032.697][INFO]: Done waiting for pending navigations. Status: ok
[1648808033.042][INFO]: Waiting for pending navigations...
[1648808034.786][INFO]: Done waiting for pending navigations. Status: ok
[1648808034.786][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE Navigate
[1648808034.791][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND ExecuteScript {
   "args": [ "/docs/api" ],
   "script": "return ((href) => {\n                    if (!href) {\n                        throw new Error('\"href\" parameter is required');\n                    }\n                    return document.querySelector(`a[href='${href}']`);\n                }).apply(null, arguments)"
}
[1648808034.793][INFO]: Waiting for pending navigations...
[1648808034.794][INFO]: Done waiting for pending navigations. Status: ok
[1648808034.797][INFO]: Waiting for pending navigations...
[1648808034.799][INFO]: Done waiting for pending navigations. Status: ok
[1648808034.799][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE ExecuteScript {
   "element-6066-11e4-a52e-4f735466cecf": "bfc36fa7-6a0c-4aad-8eac-3667790d0b31"
}
[1648808034.811][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND Refresh {

}
[1648808034.811][INFO]: Waiting for pending navigations...
[1648808034.813][INFO]: Done waiting for pending navigations. Status: ok
[1648808034.818][INFO]: Waiting for pending navigations...
[1648808034.821][INFO]: Done waiting for pending navigations. Status: ok
[1648808034.821][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE Refresh
[1648808034.827][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND ClickElement {
   "id": "bfc36fa7-6a0c-4aad-8eac-3667790d0b31"
}
[1648808034.827][INFO]: Waiting for pending navigations...
[1648808034.828][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.032][INFO]: Waiting for pending navigations...
[1648808035.512][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.516][INFO]: Waiting for pending navigations...
[1648808035.517][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.517][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE ClickElement ERROR stale element reference: element is not attached to the page document
  (Session info: chrome=100.0.4896.60)
[1648808035.522][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND ExecuteScript {
   "args": [  ],
   "script": "return ((href) => {\n                    if (!href) {\n                        throw new Error('\"href\" parameter is required');\n                    }\n                    return document.querySelector(`a[href='${href}']`);\n                }).apply(null, arguments)"
}
[1648808035.522][INFO]: Waiting for pending navigations...
[1648808035.523][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.528][INFO]: Waiting for pending navigations...
[1648808035.528][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.529][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE ExecuteScript ERROR javascript error: "href" parameter is required
  (Session info: chrome=100.0.4896.60)
[1648808035.530][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND ExecuteScript {
   "args": [  ],
   "script": "return ((href) => {\n                    if (!href) {\n                        throw new Error('\"href\" parameter is required');\n                    }\n                    return document.querySelector(`a[href='${href}']`);\n                }).apply(null, arguments)"
}
[1648808035.530][INFO]: Waiting for pending navigations...
[1648808035.531][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.533][INFO]: Waiting for pending navigations...
[1648808035.533][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.533][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE ExecuteScript ERROR javascript error: "href" parameter is required
  (Session info: chrome=100.0.4896.60)
[1648808035.535][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND ExecuteScript {
   "args": [  ],
   "script": "return ((href) => {\n                    if (!href) {\n                        throw new Error('\"href\" parameter is required');\n                    }\n                    return document.querySelector(`a[href='${href}']`);\n                }).apply(null, arguments)"
}
[1648808035.535][INFO]: Waiting for pending navigations...
[1648808035.614][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.615][INFO]: Waiting for pending navigations...
[1648808035.616][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.616][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE ExecuteScript ERROR javascript error: "href" parameter is required
  (Session info: chrome=100.0.4896.60)
[1648808035.617][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND ExecuteScript {
   "args": [  ],
   "script": "return ((href) => {\n                    if (!href) {\n                        throw new Error('\"href\" parameter is required');\n                    }\n                    return document.querySelector(`a[href='${href}']`);\n                }).apply(null, arguments)"
}
[1648808035.617][INFO]: Waiting for pending navigations...
[1648808035.618][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.619][INFO]: Waiting for pending navigations...
[1648808035.620][INFO]: Done waiting for pending navigations. Status: ok
[1648808035.620][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE ExecuteScript ERROR javascript error: "href" parameter is required
  (Session info: chrome=100.0.4896.60)
[1648808035.630][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND GetLog {
   "type": "browser"
}
[1648808035.631][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE GetLog [ {
   "level": "WARNING",
   "message": "https://webdriver.io/ 29 Unrecognized feature: 'web-share'.",
   "source": "other",
   "timestamp": 1.648808033212e+12
}, {
   "level": "WARNING",
   "message": "https://webdriver.io/ 29 Unrecognized feature: 'web-share'.",
   "source": "other",
   "timestamp": 1.648808034992e+12
} ]
[1648808035.631][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND Screenshot {

}
[1648808035.631][INFO]: Waiting for pending navigations...
[1648808035.632][INFO]: Done waiting for pending navigations. Status: ok
[1648808036.331][INFO]: Waiting for pending navigations...
[1648808036.333][INFO]: Done waiting for pending navigations. Status: ok
[1648808036.334][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE Screenshot "iVBORw0KGgoAAAANSUhEUgAADSAAAAZMCAYAAACkCM3oAAABKmlDQ1BTa2lhAAAokX2RO0sDQRRGjyGND2xUYmGxhaCNuibBB2hhliQIVolCtNtsliAmcdls0Fqx0h9hbW0p1ukFK1utBcU2fpMUG5B4hztz5rt3HncGEvPIkjY0W1FYKuasyvGJxZC5XjtgtI3Bz..."
[1648808036.420][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND GetLogTypes {

}
[1648808036.420][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE GetLogTypes [ "browser", "driver" ]
[1648808036.422][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND GetLog {
   "type": "driver"
}
[1648808036.423][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE GetLog [  ]
[1648808036.423][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND GetLog {
   "type": "browser"
}
[1648808036.424][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE GetLog [  ]
[1648808036.425][INFO]: [74e99ee11edc2373110d8d6076d1bce3] COMMAND Quit {

}
[1648808036.528][INFO]: [74e99ee11edc2373110d8d6076d1bce3] RESPONSE Quit
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!

Any contributions that resolves the bug are highly appreciated. Please take a look into our [contribution guidelines](https://webdriver.io/docs/contribute) and let us know if you have any questions. Cheers!

**Comment by chrisbottin** [comment]
I have investigated this issue and here is naive approach to fix the bug...

[Bug_Fix_for_#8176.patch.zip](https://github.com/webdriverio/webdriverio/files/8396854/Bug_Fix_for_.8176.patch.zip)

I'm sure this won't be acceptable but it should provide a base and insight to where the issue lies.

Also the fix should be also applied to browser.custom$$, element.custom$$ and element.custom$.

**Comment by christian-bromann** [comment]
> I'm sure this won't be acceptable

Why would you think that?

How about you raise a PR and we can continue our convo there?

**Comment by chrisbottin** [comment]
Ok, will do.

**Comment by christian-bromann** [comment]
A patch for this has been merged and released with `v7.19.4`.

Thanks @chrisbottin for your contribution on this. Really appreciated.
